rel/overlay/etc/default.ini

```diff
@@ -930,3 +930,99 @@ url = {{nouveau_url}}
 ;max_objects = 10000
 ;max_idle = 600000
 ;enable = true
+
+[couch_scanner]
+; How often to check for configuration changes and start/stop plugins
+;interval_sec = 5
+
+; Minimum time to force a plugin to wait before running again after a crash
+;min_penalty_sec = 30
+
+; Maximum time to force a plugin to wait after repeated crashes (8 hours default)
+;max_penalty_sec = 28800
+
+; If plugin runs successfully without crashing for this long, reset its
+; repeated error count
+;heal_threshold_sec = 300
+
+; Database processing rate limit per second. This will also be the
+; rate at which design documents are fetched. The rate is shared
+; across all running plugins.
```

Comment:
Currently the rate limiter bucket refill period is driven by the `couch_scanner_server` interval period (= 5 second) so we fill the bucket with the "5 seconds worth of tokens", that provides some batching and allows bursting for a bit, but also wonder if it's too much bursting and maybe a dedicated rate limiter gen_server refilling the bucket every 1 or 2 seconds would be better and then the interval in couch_scanner_server can be increased as it's not tied to the rate limiter (it can 10-15 seconds or so).

Reply:
It's hard to know if it's necessary without seeing how it works in production, but a simple dedicated gen_server that just refills tokens might be easier to reason about.

Reply:
Yeah reasoning part is probably what makes sense to do it. There is already a separate module for the rate limiter shouldn't be hard to plug it into the supervisor.

Reply:
I updated the pr with a fixup to switch to a separate rate limiter gen_server